buildkite/resource_pipeline.go

```diff
@@ -61,479 +71,570 @@ type TeamPipelineNode struct {
-					"exists and is supported by Terraform. See https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle.",
-				Type:        schema.TypeBool,
+			"deletion_protection": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
 				Description: "If set to 'true', deletion of a pipeline via `terraform destroy` will fail, until set to 'false'.",
 			},
-			"maximum_timeout_in_minutes": {
+			"maximum_timeout_in_minutes": schema.Int64Attribute{
 				Computed: true,
 				Optional: true,
 				Default:  nil,
-				Type:     schema.TypeInt,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
 			},
-			"description": {
+			"description": schema.StringAttribute{
 				Computed: true,
 				Optional: true,
-				Type:     schema.TypeString,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"name": {
+			"name": schema.StringAttribute{
 				Required: true,
-				Type:     schema.TypeString,
 			},
-			"repository": {
+			"repository": schema.StringAttribute{
 				Required: true,
-				Type:     schema.TypeString,
 			},
-			"skip_intermediate_builds": {
+			"skip_intermediate_builds": schema.BoolAttribute{
 				Computed: true,
 				Optional: true,
-				Type:     schema.TypeBool,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"skip_intermediate_builds_branch_filter": {
+			"skip_intermediate_builds_branch_filter": schema.StringAttribute{
 				Computed: true,
 				Optional: true,
-				Type:     schema.TypeString,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"slug": {
+			"slug": schema.StringAttribute{
 				Computed: true,
-				Type:     schema.TypeString,
 			},
-			"steps": {
+			"steps": schema.StringAttribute{
 				Optional: true,
-				Default:  defaultSteps,
-				Type:     schema.TypeString,
+				Computed: true,
+				Default:  stringdefault.StaticString(defaultSteps),
+			},
+			"tags": schema.SetAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
 			},
-			"team": {
-				Type:       schema.TypeSet,
-				Optional:   true,
-				ConfigMode: schema.SchemaConfigModeAttr,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"slug": {
+			"webhook_url": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"badge_url": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"team": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"slug": schema.StringAttribute{
```

Comment:
`team` object uses `id` and `access_level`, why not accept that? Makes it easier to have the inheritance in the config anyway.

Reply:
The slug and access_level matches the SDKv2 version of this resource. I've only added team_id and pipeline_team_id as computed to make the code simpler. Not sure why I didn't originally do this. It'll be transparent to the user. It'll automatically update those fields in state when they run a terraform refresh but it makes the code much easier to work with.

And this will be going away soon as well anyway

Reply:
Yeah, sweet. I think we're moving away from anything _but_ ID for, though we may support additional stuff for data source.